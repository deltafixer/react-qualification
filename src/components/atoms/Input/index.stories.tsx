import React from "react";
import { action } from "@storybook/addon-actions";
import Input from "./";

export default {
  component: Input,
  title: "Input",
  // excludes dummy objects
  excludeStories: /.*Data$/,
};

export const actionsData = {
  onChange: action("onChange"),
};

export const Default = () => <Input {...actionsData} />;

export const Autofocused = () => <Input autoFocus {...actionsData} />;

export const WithCustomPlaceholder = () => (
  <Input placeholder={"Custom text"} {...actionsData} />
);
