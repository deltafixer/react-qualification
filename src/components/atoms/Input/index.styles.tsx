import styled from "styled-components";

export const StyledInput = styled.input`
  height: 50px;
  font-size: 24px;
  border-radius: 6px;
  outline: none;
  padding: 0 12px;
`;
