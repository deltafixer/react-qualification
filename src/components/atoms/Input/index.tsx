import React from "react";

type InputProps = {
  type: string;
  autoFocus: boolean;
  placeholder: string;
  onChange: Function;
};

const Input = ({
  type,
  autoFocus,
  placeholder,
  onChange,
}: InputProps): JSX.Element => {
  return (
    <input
      type={type}
      autoFocus={autoFocus}
      placeholder={placeholder}
      onChange={(event) => onChange && onChange(event.target.value)}
    />
  );
};

Input.defaultProps = {
  type: "text",
  autoFocus: false,
  placeholder: "aaa",
  onChange: null,
};

export default Input;
