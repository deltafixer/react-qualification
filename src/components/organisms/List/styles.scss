.items-container {
  display: flex;
  flex-direction: column;

  div {
    margin-bottom: 3px;

    &:last-of-type {
      margin-bottom: 0;
    }
  }
}
