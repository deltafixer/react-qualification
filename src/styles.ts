import "./App.scss";
import "./components/atoms/Input/styles.scss";
import "./components/atoms/Spinner/styles.scss";
import "./components/organisms/List/styles.scss";
