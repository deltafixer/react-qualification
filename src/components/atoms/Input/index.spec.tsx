import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Input from "./index";
import { StyledInput } from "./index.styles";

let wrapper: ShallowWrapper;

const props = {
  onChange: jest.fn(),
};

describe("Input component tests", () => {
  beforeAll(() => {
    wrapper = shallow(<Input {...props} />);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("triggers callback when 'change' event catched on input field", () => {
    wrapper.find(StyledInput).simulate("change", {
      target: {
        value: "some text",
      },
    });

    expect(props.onChange).toHaveBeenCalledTimes(1);
  });
});
