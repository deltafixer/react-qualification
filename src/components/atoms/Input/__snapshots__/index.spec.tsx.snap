// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests renders correctly 1`] = `"<styled.input type=\\"text\\" autoFocus={false} placeholder=\\"\\" onChange={[Function: onChange]} />"`;
